core/lib/object_store/src/raw.rs

```diff
@@ -88,7 +88,7 @@ impl error::Error for ObjectStoreError {
 ///
 /// [`StoredObject`]: crate::StoredObject
 #[async_trait]
-pub trait ObjectStore: fmt::Debug + Send + Sync {
+pub trait ObjectStore: 'static + fmt::Debug + Send + Sync {
```

Comment:
It's not really required, I just noticed that it's lacking and thought that I'll add it. Non-static async traits are weirdly specific, and I think it makes sense to restrict it here just to prevent some poor soul from a lot of pain in the future.